### PR TITLE
Clean-up Dashoboard Blocks

### PR DESCRIPTION
**Pull request by kucrut**
This provides DashboardBlocks.withData() HOC so that the user components don't need to care about loading state or error and just render whatever they need to when they get the `data` they need.

### CODE REVIEW

File: src/components/Dashboard-Block.js [hunk 1]
```diff
@@ -23,28 +26,56 @@ class DashboardBlock extends Component {
...
+		class WithData extends Component {
+			render() {
+				const { error, loading, ...rest } = this.props;
 
-export default DashboardBlock;
+				if ( loading ) {
+					return <p>Loading...</p>;
+				}
+
+				if ( error ) {
```
Comment (kucrut) [inline src/components/Dashboard-Block.js]: Somehow, `error` is always `true` even when the request returns 404, and the error object is received as `data`.

**Comment by kucrut** [comment]
@roborourke Don't merge yet, as I still need to deal with error (see https://github.com/humanmade/platform-ui/pull/37/files#r233469027 ). Though, I would like your thoughts on the `DashboardBlock.withData()` HOC.

File: src/components/Dashboard-Block.js (hunk 1 above)
Comment (roborourke) [reply src/components/Dashboard-Block.js]: That's not ideal! It definitely should be passed as `data`, maybe an issue with the cache lookup

Comment (kucrut) [reply src/components/Dashboard-Block.js]: Resolved via https://github.com/humanmade/platform-ui/pull/37/commits/c32279773fec30149fd4433ba162bfa24ec96422

**Comment by roborourke** [comment]
Tested out and looks 👍, made a small tweak to the deploy item styles

**Pull request merged**